spec/acceptance/conf_setting_spec.rb

```diff
@@ -1,33 +1,47 @@
+# frozen_string_literal: true
+
 require 'spec_helper_acceptance'
 
-tmpdir = setup_test_directory
+tmpdir = default.tmpdir('tmp')
 
 describe 'hocon_setting resource' do
   after :all do
-    run_shell("rm #{tmpdir}/*.conf", acceptable_exit_codes: [0, 1, 2])
+    shell("rm #{tmpdir}/*.conf", acceptable_exit_codes: [0, 1, 2])
   end
 
   shared_examples 'has_content' do |path, pp, content|
     before :all do
-      run_shell("rm #{path}", expect_failures: true)
+      shell("rm #{path}", acceptable_exit_codes: [0, 1, 2])
+    end
+
+    after :all do
+      shell("cat #{path}", acceptable_exit_codes: [0, 1, 2])
+      shell("rm #{path}", acceptable_exit_codes: [0, 1, 2])
     end
 
     it 'applies the manifest twice' do
-      idempotent_apply(pp)
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
     end
 
     describe file(path) do
       it { is_expected.to be_file }
       # XXX Solaris 10 doesn't support multi-line grep
-      it("should contain #{content}") {
+
+      it("contains #{content}", unless: fact('osfamily') == 'Solaris') {
```

Comment:
Solaris isn't in our supported operating systems. I think it's safe to drop this.